src/containerapp/azext_containerapp/_compose_ported.py

```diff
@@ -0,0 +1,235 @@
+    """
+    if not hasattr(service, "models") or service.models is None:
+        return {}
+    if not isinstance(service.models, dict):
+        return {}
+    return service.models
+
+
+def detect_service_type(service) -> str:
+    """Ported from 092e028c556c5d98c06ea1a337c26b97fe00ce59.
+
+    Classify a compose service so that compose processing can customise
+    behaviour for MCP gateway, model-runner, agent, or generic services.
+    """
+    service_name = service.name.lower() if hasattr(service, "name") else ""
+    image_name = service.image.lower() if getattr(service, "image", None) else ""
+    command_str = ""
+    if getattr(service, "command", None) is not None:
+        command = service.command
+        command_str = command.command_string().lower() if hasattr(command, "command_string") else str(command).lower()
+
+    if "mcp-gateway" in service_name or "mcp-gateway" in image_name or "--servers" in command_str:
+        return "mcp-gateway"
+    if "model-runner" in service_name or "model-runner" in image_name:
+        return "model-runner"
+    if hasattr(service, "models") and service.models:
+        return "agent"
+    if hasattr(service, "depends_on") and service.depends_on:
+        depends_on_iter = service.depends_on
+        if isinstance(depends_on_iter, dict):
+            depends_on_iter = depends_on_iter.keys()
+        for dependency in depends_on_iter:
+            dep_str = str(dependency).lower()
+            if "mcp-gateway" in dep_str or "model-runner" in dep_str:
+                return "agent"
+    return "generic"
+
+
+def parse_mcp_servers_from_command(service) -> List[Dict[str, object]]:
+    """Ported from 092e028c556c5d98c06ea1a337c26b97fe00ce59.
+
+    Inspect the MCP gateway command line for ``--servers``/``--tools`` flags
+    and return a normalised list of server definitions.
+    """
+    if getattr(service, "command", None) is None:
+        return []
+
+    command = service.command
+    command_str = command.command_string() if hasattr(command, "command_string") else str(command)
+    command_parts = command_str.split()
+
+    servers: List[str] = []
+    tools: List[str] = []
+    for idx, part in enumerate(command_parts):
+        if part == "--servers" and idx + 1 < len(command_parts):
+            servers = [item.strip() for item in command_parts[idx + 1].split(",") if item.strip()]
+        if part == "--tools" and idx + 1 < len(command_parts):
+            tools = [item.strip() for item in command_parts[idx + 1].split(",") if item.strip()]
+
+    return [
+        {
+            "name": server_name,
+            "server_type": server_name,
+            "tools": tools if tools else ["*"],
+            "image": f"docker/mcp-server-{server_name}",
+            "resources": {"cpu": "0.5", "memory": "1.0"},
+        }
+        for server_name in servers
+    ]
+
+
+def should_deploy_model_runner(compose_yaml: Dict, parsed_compose_file) -> bool:
```

Comment:
Do we have any changes in these functions? Do you mean some of them changed and some are just copy?
Recommend to import for `azure-cli` directly